tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -557,51 +506,59 @@ spec:
       mode: PERMISSIVE`,
 					expected: []expect{
 						{
-							port:              8085,
-							schema:            protocol.HTTP,
+							port:              &echo.Port{ServicePort: 8085, Protocol: protocol.HTTP},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      false,
 						},
 						{
-							port:              8086,
-							schema:            protocol.HTTP,
+							port:              &echo.Port{ServicePort: 8086, Protocol: protocol.HTTP},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      true,
 						},
 						{
-							port:              8087,
-							schema:            protocol.TCP,
+							port:              &echo.Port{ServicePort: 8087, Protocol: protocol.TCP},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      false,
 						},
 						{
-							port:              8088,
-							schema:            protocol.TCP,
+							port:              &echo.Port{ServicePort: 8088, Protocol: protocol.TCP},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      true,
 						},
 						{
-							port:              8089,
-							schema:            protocol.HTTPS,
+							port:              &echo.Port{ServicePort: 8089, Protocol: protocol.HTTPS},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      false,
 						},
 						{
-							port:              8084,
-							schema:            protocol.HTTPS,
+							port:              &echo.Port{ServicePort: 8084, Protocol: protocol.HTTPS},
 							plaintextSucceeds: true,
 							mtlsSucceeds:      true,
 						},
 					},
 				},
 			}
 
-			for _, cluster := range t.Clusters() {
-				destination := apps.A.Match(echo.Namespace(ns.Name())).GetOrFail(t, echo.InCluster(cluster))
-				destWorkload := getWorkload(destination, t).Address()
-				// Its not trivial to force mTLS to passthrough ports. To workaround this, we will
-				// set up a SE and DR that forces it
-				se := tmpl.EvaluateOrFail(t, `apiVersion: networking.istio.io/v1beta1
+			// srcFilter finds the naked app as client.
+			// TODO(slandow) replace this with built-in framework filters (blocked by https://github.com/istio/istio/pull/31565)
+			srcFilter := []echotest.SimpleFilter{func(instances echo.Instances) echo.Instances {
+				src := apps.Naked.Match(echo.Namespace(ns.Name()))
+				b := apps.B.Match(echo.Namespace(ns.Name()))
```

Comment:
nit: this is a littler hard to read, can you do `src = append(src, apps.B.Match(...`?